src/IonEventStream.ts

```diff
@@ -0,0 +1,410 @@
+                    currentEvent.set(fieldName, this.reader.numberValue());
+                    break;
+                }
+
+                default :
+                    throw new Error('Unexpected event field: ' + fieldName);
+            }
+        }
+        let eventType : IonEventType;
+        switch(currentEvent.get('event_type')){
+            case 'CONTAINER_START' :
+                eventType = IonEventType.CONTAINER_START;
+                break;
+            case 'STREAM_END' :
+                eventType = IonEventType.STREAM_END;
+                break;
+            case 'CONTAINER_END' :
+                eventType = IonEventType.CONTAINER_END;
+                break;
+            case 'SCALAR' :
+                eventType = IonEventType.SCALAR;
+                break;
+            case 'SYMBOL_TABLE' :
+                throw new Error('Symbol tables unsupported');
+        }
+        return this.eventFactory.makeEvent(
+            eventType,
+            currentEvent.get('ion_type'),
+            currentEvent.get('field_name'),
+            currentEvent.get('depth'),
+            currentEvent.get('annotations'),
+            currentEvent.get('isNull'),
+            currentEvent.get('value_text')
+        );
+    }
+
+    private parseIonType() : IonType{
+        let input : string = this.reader.stringValue().toLowerCase();
+        switch(input) {
+            case 'null' : {
+                return IonTypes.NULL;
+            }
+            case 'bool' : {
+                return IonTypes.BOOL;
+            }
+            case 'int' : {
+                return IonTypes.INT;
+            }
+            case 'float' : {
+                return IonTypes.FLOAT;
+            }
+            case 'decimal' : {
+                return IonTypes.DECIMAL;
+            }
+            case 'timestamp' : {
+                return IonTypes.TIMESTAMP;
+            }
+            case 'symbol' : {
+                return IonTypes.SYMBOL;
+            }
+            case 'string' : {
+                return IonTypes.STRING;
+            }
+            case 'clob' : {
+                return IonTypes.CLOB;
+            }
+            case 'blob' : {
+                return IonTypes.BLOB;
+            }
+            case 'list' : {
+                return IonTypes.LIST;
+            }
+            case 'sexp' : {
+                return IonTypes.SEXP;
+            }
+            case 'struct' : {
+                return IonTypes.STRUCT;
+            }
+            default : {
+                throw new Error('unexpected type: ' + input);
+            }
+
+        }
+    }
+
+    private parseAnnotations() : string[] {
+        let annotations : string[] = [];
+        if(this.reader.isNull()) {
+           return annotations;
+        } else {
+            this.reader.stepIn();
+            for(let tid = this.reader.next(); tid !== undefined; tid = this.reader.next()) {
+                annotations.push(this.reader.value());
```

Comment:
Needs to be converted to reading a list of SymbolTokens.

Reply:
noted